### ISSUE TITLE
Latest firmware missing deCONZ_Rpi_0x26370500.bin.GCF

### ISSUE THREAD

**Issue by corvy** [marthoc/docker-deconz#261]
Hello, according to the releasenotes here: https://github.com/dresden-elektronik/deconz-rest-plugin/releases/tag/V2_05_81_stable this update should include new version of firmware: 

    Firmware version 0x26370500 for ConBee I and RaspBee I.
    Firmware version 0x26650700 for ConBee II and RaspBee II.

These fix, or "mitigate multi-hop routing issues and improve network stability the latest firmware uses source routing in parallel to normal mesh routing.". These are issues I really want to fix! :) 

So I basically did the flash with a small workaround. I started the update procedure (after stopping the plugin): 

`docker run -it --rm --entrypoint "/firmware-update.sh" --privileged --cap-add=ALL -v /dev:/dev -v /lib/modules:/lib/modules -v /sys:/sys marthoc/deconz:testing-armhf`

After the container has started, I stopped when it prompts for device. Then I connected to the container shell, I used Portainer, but you could open a new shell and connect with exec (https://docs.docker.com/engine/reference/commandline/exec/). I then downloaded the newest firmware manually. 

For RasbBee:
`cd /usr/share/deCONZ/firmware
curl https://deconz.dresden-elektronik.de/deconz-firmware/deCONZ_Rpi_0x26370500.bin.GCF --output deCONZ_Rpi_0x26370500.bin.GCF`

After this I continue in the other window to complete the flashing process. Obs! Check your hardware to get the right firmware. https://deconz.dresden-elektronik.de/deconz-firmware/. 

This issue is to raise the fact that we are missing the latest firmwares in the latest release. Could we get these added? Thank you!

![Untitled](https://user-images.githubusercontent.com/138292/94335422-e13ccb80-ffdb-11ea-9e6c-559ad8d2cd9d.png)

<img width="392" alt="Screenshot 2020-09-26 at 09 34 16" src="https://user-images.githubusercontent.com/138292/94335428-ebf76080-ffdb-11ea-9309-12bd113e622c.png">

**Comment by manup** [comment]
The files are currently not part of the deCONZ package, they will be included in the next beta version arriving on the 15th October. As as imple workaround it might also be possible to just mount a `/firmware` directory containing the files into `/usr/share/deCONZ/firmware` in the container.

**Comment by sorryusernameisalreadytaken** [comment]
@corvy THX for your cmd's. I did it this way:

Here is my funky crazy raspberrypi4 (raspbian) on-liner: :D
```
docker run -it --rm --entrypoint "/bin/bash" --privileged --cap-add=ALL -v /dev:/dev -v /lib/modules:/lib/modules -v /sys:/sys marthoc/deconz:testing-armhf "-c" "read -p 'Full firmware file with ending GCF:' fwname && echo Try to download file \$fwname && curl -o /usr/share/deCONZ/firmware/\$fwname http://deconz.dresden-elektronik.de/deconz-firmware/\$fwname && echo && echo This should be your md5sum: && curl http://deconz.dresden-elektronik.de/deconz-firmware/\$fwname.md5 && echo Please compare with local md5sum: && md5sum /usr/share/deCONZ/firmware/\$fwname && sed 's/read -p \"File Name : \" fileName/read -e -p \"File Name : \" -i \"'\$fwname'\" fileName/g' firmware-update.sh >fwupd.sh && chmod +x fwupd.sh && ./fwupd.sh"
```
> 
> 
> The files are currently not part of the deCONZ package, they will be included in the next beta version arriving on the 15th October. As as imple workaround it might also be possible to just mount a `/firmware` directory containing the files into `/usr/share/deCONZ/firmware` in the container.

How can we improve this for future updates?

**Comment by PeterRemnemark** [comment]
> @corvy THX for your cmd's. I did it this way:
> 
> Here is my funky crazy raspberrypi4 (raspbian) on-liner: :D
> 
> ```
> docker run -it --rm --entrypoint "/bin/bash" --privileged --cap-add=ALL -v /dev:/dev -v /lib/modules:/lib/modules -v /sys:/sys marthoc/deconz:testing-armhf "-c" "fwdir='/usr/share/deCONZ/firmware/' && fwurl='https://deconz.dresden-elektronik.de/deconz-firmware/' && rm \$fwdir/* && read -p '--Press any key to get the latest 10 firmware-files--' && fwsonln=\$(curl -s \$fwurl'?C=M;O=D'|sed -n 's/.*href=\"\([^\"]*\).*/\1/p'|grep -m 10 -E '^deCONZ_.*_0x.+[a-f0-9].bin.GCF\$') && echo \$fwsonln|tr \" \" \"\n\" && newfw=\$(echo \$fwsonln|tr \" \" \"\n\"|grep -m 1 GCF) && read -e -p 'Enter Firmware file (with ending GCF)?:' -i \$newfw fwname && curl -s -o \$fwdir\$fwname \$fwurl\$fwname && echo 'Remote md5 checksum:' && curl -s \$fwurl\$fwname.md5 && echo 'Local md5 checksum:' && md5sum \$fwdir\$fwname && read -p '--Both md5 checksum must be the same! If yes, press any key--' && sed 's/read -p \"File Name : \" fileName/read -e -p \"File Name : \" -i \"'\$fwname'\" fileName/g' firmware-update.sh >fwupd.sh && chmod +x fwupd.sh && ./fwupd.sh"
> ```
> 
> ![extende_raspbee_updater1liner](https://user-images.githubusercontent.com/12913436/94579761-77d9e880-0279-11eb-8f6a-11844552f82c.gif)
> 
> > The files are currently not part of the deCONZ package, they will be included in the next beta version arriving on the 15th October. As as imple workaround it might also be possible to just mount a `/firmware` directory containing the files into `/usr/share/deCONZ/firmware` in the container.
> 
> How can we improve this for future updates?

This was awesome! Worked for me, just had to rename the image name and worked at a Synology instance of docker. I do agree that an feature to put the fw in a local folder (or as you have implemented it, download it on the fly) would be very handy.

**Comment by Jo-Be-Co** [comment]
> @corvy THX for your cmd's. I did it this way:
> 
> Here is my funky crazy raspberrypi4 (raspbian) on-liner: :D

Hi, I used some of your one-liner ideas to create an update on the firmware scripts. #267

**Comment by sorryusernameisalreadytaken** [comment]
> 
> 
> > @corvy THX for your cmd's. I did it this way:
> > Here is my funky crazy raspberrypi4 (raspbian) on-liner: :D
> 
> Hi, I used some of your one-liner ideas to create an update on the firmware scripts. #267

Thats cool.
I gave my 2 cents to it ;-)

**Comment by marthoc** [comment]
Thanks to @Jo-Be-Co, this should now be addressed in version 2.05.83 of the image, now available on Docker Hub. In the flashing script you should now be able to specify the filename deCONZ_Rpi_0x26370500.bin.GCF and it will be downloaded from Dresden Elektronik's website! Many thanks to @Jo-Be-Co for implementing!
